smartsim/entity/model.py

```diff
@@ -162,39 +230,53 @@ def colocate_db(
 
         :param port: port to use for orchestrator database, defaults to 6379
         :type port: int, optional
+        :param ifname: interface to use for orchestrator, defaults to "lo"
+        :type ifname: str, optional
         :param db_cpus: number of cpus to use for orchestrator, defaults to 1
         :type db_cpus: int, optional
         :param limit_app_cpus: whether to limit the number of cpus used by the app, defaults to True
         :type limit_app_cpus: bool, optional
-        :param ifname: interface to use for orchestrator, defaults to "lo"
-        :type ifname: str, optional
         :param debug: launch Model with extra debug information about the co-located db
         :type debug: bool, optional
         :param kwargs: additional keyword arguments to pass to the orchestrator database
         :type kwargs: dict, optional
 
         """
+
+        tcp_options = {
+            "port":port,
+            "ifname":ifname
+        }
+        common_options = {
+            "cpus":db_cpus,
+            "limit_app_cpus":limit_app_cpus,
+            "debug":debug
+        }
+        self._set_colocated_db_settings( tcp_options, common_options, **kwargs)
+
+    def _set_colocated_db_settings(self, connection_options, common_options, **kwargs):
+        """
+        Ingest the connection-specific options (UDS/TCP) and set the final settings
+        for the co-located database
+        """
+
         if hasattr(self.run_settings, "mpmd") and len(self.run_settings.mpmd) > 0:
             raise SSUnsupportedError(
                 "Models co-located with databases cannot be run as a mpmd workload"
             )
 
         if hasattr(self.run_settings, "_prep_colocated_db"):
-            self.run_settings._prep_colocated_db(db_cpus)
+            self.run_settings._prep_colocated_db(common_options['db_cpus'])
 
         # TODO list which db settings can be extras
-        colo_db_config = {
-            "port": int(port),
-            "cpus": int(db_cpus),
-            "interface": ifname,
-            "limit_app_cpus": limit_app_cpus,
-            "debug": debug,
-            # redisai arguments for inference settings
-            "rai_args": {
-                "threads_per_queue": kwargs.get("threads_per_queue", None),
-                "inter_op_parallelism": kwargs.get("inter_op_parallelism", None),
-                "intra_op_parallelism": kwargs.get("intra_op_parallelism", None),
-            },
+        colo_db_config = {}
+        colo_db_config.update(connection_options)
+        colo_db_config.update(common_options)
+        # redisai arguments for inference settings
+        colo_db_config['rai_args'] = {
+            "threads_per_queue": kwargs.get("threads_per_queue", None),
+            "inter_op_parallelism": kwargs.get("inter_op_parallelism", None),
+            "intra_op_parallelism": kwargs.get("intra_op_parallelism", None),
```

Comment:
I know this was here prior to this PR, so feel free to ignore this comment, but is there any reason these cannot become named kwargs?

Reply:
I think it might be just a strictly organiztional thing? I'm a little loathe to start pulling in this thread since since I'd rather we do a refactor of the entire handling of the **kwargs as opposed to just focusing on this one instance.

Reply:
Very fair! I could easily see this being the subject of a separate PR. Consider this resolved!!